### PR TITLE
Improve date filter

### PR DESCRIPTION
**Pull request by maikschneider**
closes #44 

## Summary by CodeRabbit

* **New Features**
  * Added three new date/datetime fields to extended-news content blocks for flexible date storage and display.

* **Bug Fixes**
  * Fixed date filtering so comparisons work correctly across different stored date/datetime formats and DB column types.

* **Tests**
  * Added end-to-end tests validating date filter behavior (operators and boundary cases) for all new date field variants.

### CODE REVIEW

File: Classes/Controller/AbstractBackendController.php [hunk 1]
```diff
@@ -604,7 +604,27 @@ protected function addFilterConstraint(): void
...
                 }
                 if (isset($data['dataType']) && $data['dataType'] === 'date') {
-                    $data['value'] = strtotime($data['value']);
+                    $dbType = $GLOBALS['TCA'][$this->getTableName()]['columns'][$field]['config']['dbType'] ?? '';
+                    $date = date('Y-m-d', strtotime($data['value']));
+                    if ($dbType === 'date' || $dbType === 'datetime') {
+                        $leftExpr = 'IFNULL(DATE(t1.' . $field . '), "")';
+                    } else {
+                        $leftExpr = 'IFNULL(DATE(FROM_UNIXTIME(t1.' . $field . ')), "")';
+                    }
```
[the remark this answers is not in the document]
Comment (maikschneider) [reply Classes/Controller/AbstractBackendController.php:613]: When having empty values, I like the date to be treated as "0" - so being in the search results for `lt` and `not` is okay
